src/Core/System/Language/Command/LanguageChangeDefaultCommand.php

```diff
@@ -0,0 +1,300 @@
+            (new Criteria([Defaults::LANGUAGE_SYSTEM]))
+                ->addAssociation('locale'),
+            $context
+        )->first();
+
+        $currentLocaleId = $currentDefaultLanguage->getLocaleId();
+        $currentLocale = $currentDefaultLanguage->getLocale();
+        if (!$currentLocale) {
+            $output->writeln('<error>couldn\'t find current locale</error>');
+
+            return 1;
+        }
+        $newDefaultLocaleId = $newLocale->getId();
+
+        // locales match -> do nothing.
+        if ($currentLocaleId === $newDefaultLocaleId) {
+            $output->writeln(sprintf('<comment>nothing todo %s is already the system default language</comment>', $localeCode));
+
+            return 0;
+        }
+
+        $newDefaultLanguageId = $this->languageRepository->searchIds(
+            (new Criteria())
+                ->addFilter(new EqualsFilter('localeId', $newDefaultLocaleId)),
+            $context
+        )->firstId();
+
+        if (!$newDefaultLanguageId) {
+            $newDefaultLanguageId = $this->createNewLanguageEntry($newLocale, $context);
+        }
+
+        $this->swapDefaultLanguageId($newDefaultLanguageId);
+
+        $output->writeln(sprintf('<info>system default language changed to %s</info>', $newLocale->getCode()));
+
+        return 0;
+    }
+
+    private function getLocales(): array
+    {
+        $context = new Context(new SystemSource());
+        /** @var LocaleEntity[] $locales */
+        $locales = $this->localeRepository->search(
+            (new Criteria())
+                ->addAssociation('translations')
+                ->addSorting(new FieldSorting('translations.name')),
+            $context
+        )->getEntities();
+        $localeList = [];
+
+        foreach ($locales as $locale) {
+            if (!array_key_exists($locale->getName(), $localeList)) {
+                $localeList[$locale->getName()] = [];
+            }
+            $localeList[$locale->getName()][] = $locale->getCode();
+        }
+
+        return $localeList;
+    }
+
+    private function createNewLanguageEntry(LocaleEntity $locale, Context $context): string
+    {
+        $ids = $this->languageRepository->create([[
+            'localeId' => $locale->getId(),
+            'name' => $locale->getTranslation('name'),
+            'translationCodeId' => $locale->getId(),
+        ]], $context)
+            ->getPrimaryKeys(LanguageDefinition::ENTITY_NAME);
+
+        return array_shift($ids);
+    }
+
+    /**
+     * @throws DoctrineDriverException
+     * @throws Exception
+     */
+    private function swapDefaultLanguageId(string $newLanguageId): void
+    {
+        // $this->connection->executeStatement('SET foreign_key_checks = 0;');
+        $stmt = $this->connection->prepare(
+            'UPDATE language
+             SET id = :newId
+             WHERE id = :oldId'
+        );
+
+        $tmpId = Uuid::randomBytes();
+
+        // assign new uuid to old DEFAULT
+        $stmt->execute([
```

Comment:
This may destroy the translation chain, as it is not enforced that all translations are defined in the new default language, maybe we can add a check that verifies that all  translations also exist in the new language